src/test/java/com/hotels/bdp/circustrain/bigquery/client/BigQueryMetastoreClientTest.java

```diff
@@ -42,60 +40,45 @@
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
 
-import com.google.api.gax.paging.Page;
-import com.google.cloud.bigquery.BigQuery;
-import com.google.cloud.bigquery.Dataset;
-import com.google.cloud.bigquery.Job;
-import com.google.cloud.bigquery.JobStatus;
-import com.google.cloud.bigquery.Schema;
-import com.google.cloud.bigquery.TableDefinition;
-import com.google.cloud.storage.Blob;
-import com.google.cloud.storage.Storage;
-
 import com.hotels.bdp.circustrain.bigquery.extraction.service.ExtractionService;
 import com.hotels.bdp.circustrain.bigquery.table.service.TableServiceFactory;
 import com.hotels.bdp.circustrain.bigquery.table.service.partitioned.PartitionedTableService;
-import com.hotels.bdp.circustrain.bigquery.table.service.unpartitioned.UnpartitionedTableService;
 import com.hotels.bdp.circustrain.bigquery.util.BigQueryMetastore;
-import com.hotels.bdp.circustrain.bigquery.util.TableNameFactory;
 
 @RunWith(MockitoJUnitRunner.class)
 public class BigQueryMetastoreClientTest {
 
   private final String databaseName = "database";
   private final String tableName = "table";
 
-  private @Mock BigQuery bigQuery;
   private @Mock ExtractionService extractionService;
   private @Mock TableServiceFactory factory;
   private @Mock PartitionedTableService partitionedTableService;
-  private @Mock UnpartitionedTableService unpartitionedTableService;
-  private @Mock HiveTableCache cache;
-  private @Mock Job job;
   private @Mock BigQueryMetastore bigQueryMetastore;
-  private @Mock JobStatus jobStatus;
-  private @Mock Dataset dataset;
   private @Mock com.google.cloud.bigquery.Table table;
-  private @Mock TableDefinition tableDefinition;
-  private @Mock Storage storage;
-  private @Mock Blob blob;
-  private @Mock Page<Blob> blobs;
 
   private BigQueryMetastoreClient bigQueryMetastoreClient;
+  private List<Partition> partitions = new ArrayList<>();
+  private final HiveTableCache cache = new HiveTableCache();
   private final Table hiveTable = new Table();
 
   @Before
   public void init() throws IOException {
     hiveTable.setTableName(tableName);
     hiveTable.setDbName(databaseName);
     bigQueryMetastoreClient = new BigQueryMetastoreClient(bigQueryMetastore, extractionService, cache, factory);
+
+    for (int i = 0; i < 10; ++i) {
```

Comment:
minor but `i++` is the more common form so use that unless there's a really good reason not to.